### PR TITLE
Add optional arguments to to_pandas and flatten

### PR DESCRIPTION
**Pull request by schaefed**
In order to represent compound flags in `saqc` using instances of `DictOfSeries`/`DictOfPandas` instead of `DataFrames` we need a bit more flexibility in the `to_pandas` machinery. This is implemented here.

### CODE REVIEW

**Comment by palmb** [comment]
As i understand this, you have some need for a different name schema than the default schema (`{df-name}_{column}`). 
The scheme you present is very specific to your case and i would like you to consider a simple renaming after a `squeeze`/`to_pandas`.     

If you wish for more flexibility for the column naming, we could add a `columns` keyword to the functions in question, that takes a pandas index, to which the new columns are (re)named (this would even allow duplicate column-names) in the process of flattening.  

Finally i find the word `squeeze` misleading, because IMO nothing gets squeezed.

**Comment by palmb** [comment]
sorry i misunderstand your request. 

i'm fine with the multiindex approach, but would like the keyword to be renamed and default to `multiindex: bool = False`.

File: fancy_collections/formatting.py [hunk 1]
```diff
@@ -45,25 +47,49 @@ def key_to_string(self, key: Any, obj: Any) -> str:  # noqa
...
+            if k in front_keys:
+                self._add(k, v)
+
+        if not front_keys.intersection(back_keys):
+            self._add("...", ["..."])
+
+        for k, v in objects.items():
+            if k in back_keys and k not in front_keys:
+                self._add(k, objects[k])
+
```
Comment (palmb) [inline fancy_collections/formatting.py:82]: - [ ]  this must be merged after #1

**Review by palmb — CHANGES_REQUESTED** [review]

File: fancy_collections/formatting.py (hunk 1 above)
Comment (schaefed) [reply fancy_collections/formatting.py:82]: > i'm fine with the multiindex approach, but would like the keyword to be renamed and default to multiindex: bool = False.

Fine with that!

Comment (schaefed) [reply fancy_collections/formatting.py:82]: Done!

**Comment by palmb** [comment]
unfortunately we have merge conflicts. Could you fix them and ping me again ?

**Pull request merged**